src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -627,6 +627,189 @@ TypeSupport DataReaderImpl::type()
+    }
+    if (qos.latency_budget().duration != new_qos.latency_budget().duration)
+    {
+        qos.latency_budget() = new_qos.latency_budget();
+        qos.latency_budget().hasChanged = true;
+    }
+    if (first_time && !(qos.liveliness() == new_qos.liveliness()))
+    {
+        qos.liveliness() = new_qos.liveliness();
+        qos.liveliness().hasChanged = true;
+    }
+    if (first_time && !(qos.reliability() == new_qos.reliability()))
+    {
+        qos.reliability() = new_qos.reliability();
+        qos.reliability().hasChanged = true;
+    }
+    if (first_time && qos.ownership().kind != new_qos.ownership().kind)
+    {
+        qos.ownership() = new_qos.ownership();
+        qos.ownership().hasChanged = true;
+    }
+    if (qos.destination_order().kind != new_qos.destination_order().kind)
+    {
+        qos.destination_order() = new_qos.destination_order();
+        qos.destination_order().hasChanged = true;
+    }
+    if (qos.user_data().data_vec() != new_qos.user_data().data_vec())
+    {
+        qos.user_data() = new_qos.user_data();
+        qos.user_data().hasChanged = true;
+    }
+    if (qos.time_based_filter().minimum_separation != new_qos.time_based_filter().minimum_separation )
+    {
+        qos.time_based_filter() = new_qos.time_based_filter();
+        qos.time_based_filter().hasChanged = true;
+    }
+    if (qos.topicData().getValue() != new_qos.topicData().getValue())
+    {
+        qos.topicData() = new_qos.topicData();
+        qos.topicData().hasChanged = true;
+    }
+    if (first_time || !(qos.durabilityService() == new_qos.durabilityService()))
+    {
+        qos.durabilityService() = new_qos.durabilityService();
+        qos.durabilityService().hasChanged = true;
+    }
+    if (qos.lifespan().duration != new_qos.lifespan().duration )
+    {
+        qos.lifespan() = new_qos.lifespan();
+        qos.lifespan().hasChanged = true;
+    }
+    if (first_time && !(qos.reliable_reader_qos() == new_qos.reliable_reader_qos()))
+    {
+        qos.reliable_reader_qos() = new_qos.reliable_reader_qos();
+        qos.reliable_reader_qos().hasChanged = true;
+    }
+    if (first_time || !(qos.type_consistency() == new_qos.type_consistency()))
+    {
+        qos.type_consistency() = new_qos.type_consistency();
+        qos.type_consistency().hasChanged = true;
+    }
+    if (first_time && (qos.history().kind != new_qos.history().kind ||
+            qos.history().depth != new_qos.history().depth))
+    {
+        qos.history() = new_qos.history();
+        qos.history().hasChanged = true;
+    }
+    if (first_time && !(qos.resource_limits() == new_qos.resource_limits()))
+    {
+        qos.resource_limits() = new_qos.resource_limits();
+        qos.resource_limits().hasChanged = true;
+    }
+    if (!(qos.reader_data_lifecycle() == new_qos.reader_data_lifecycle()))
+    {
+        qos.reader_data_lifecycle() = new_qos.reader_data_lifecycle();
+        qos.reader_data_lifecycle().hasChanged = true;
```

Comment:
```suggestion
```
ReaderDataLifecycleQos doesn't need to inherit from QosPolicy as it don't have to be sent